FileUpload/src/Browse/Browse.jsx

```diff
@@ -142,24 +142,62 @@ export default class Browse extends Component {
         this.setState({ folders });
     }
 
+    renderActions(){
+        const {props} = this;
+
+        let components = [];
+        let actionTemplate = props.browseActionText;
+
+        let tokenRegex = /\{(.+?)\|(.+?)\}/;
+        while(tokenRegex.test(actionTemplate)){
+            let match = tokenRegex.exec(actionTemplate);
+
+            components.push(actionTemplate.substr(0, match.index));
+
+            let action = ((type) => {
+                switch(type.toLowerCase()){
+                    case "save":
+                        return this.onSave.bind(this);
+                    case "cancel":
+                        return this.props.onCancel;
+                    default:
+                        return null;
+                }
+            })(match[1]);
+
+            components.push(<strong onClick={action}>{match[2]}</strong>);
+
+            actionTemplate = actionTemplate.substr(match.index + match[0].length);
+        }
+
+        if(actionTemplate){
+            components.push(actionTemplate);
+        }
+
+        return components;
+    }
+
     render() {
         /* eslint-disable react/no-danger */
         return <div className="file-upload-container">
-            <h4>Folder</h4>
+            <h4>{this.props.folderText}</h4>
             <FolderPicker
+                {...this.props}
```

Comment:
I am not fully sure of using spread operator to propagate the localization and others props down to children components. I know that the other approach is more declarative and explict, but also more clear. What do you think?